lib/IDE/CodeCompletion.cpp

```diff
@@ -5167,9 +5170,16 @@ class CodeCompletionTypeContextAnalyzer {
   SmallVector<FunctionTypeAndDecl, 2> PossibleCallees;
 
   void recordPossibleType(Type ty) {
+    if (ty) {
+      ty = ty->getRValueType();
+      if (ty->hasTypeParameter())
+        if (auto env = DC->getGenericEnvironmentOfContext())
+          ty = env->mapTypeIntoContext(ty);
```

Comment:
I honestly not sure about this. But I guess the `DeclContext` should be the context where the parameter is declared. So I updated the PR so that it extract decl context from the `GenericTypeParamType` itself. @slavapestov am I doing the right thing?

Reply:
You want to use the DeclContext of the function itself.